### ISSUE TITLE
Missing SONAME in some libraries

### ISSUE THREAD

**Issue by TheCynosure** [RobotLocomotion/drake#15946]
Some of the Drake libraries are missing SONAME fields.
You can find the SONAME by running:
```
objdump -p <libname> | grep SONAME
```

The following libraries are missing SONAME's:
```
libdrake.so
libdrake_ibex.so
libdrake_marker.so
```

This causes CMake to link to an absolute path as opposed to a libname with an rpath which can be stripped during install.

According to @jwnimmer-tri the way to fix this issue is to add the `soname` link opt under linux, example: https://github.com/RobotLocomotion/drake/blob/610ec7f/tools/workspace/spdlog/package.BUILD.bazel#L66-L73

**Comment by jwnimmer-tri** [comment]
The `libdrake.so` is the most important one to fix.

We should be able fix `libdrake.so` and `libdrake_marker.so` quickly, by changing their `BUILD.bazel` files per that spdlog example.

The `libdrake_ibex.so` fix should probably wait on #13219.

@BetsyMcPhail I'm going to put this on the Build & Release board, for the `libdrake.so` fix at a minimum.

**Comment by svenevs** [comment]
- SONAME for `libdrake.so` and `libdrake_marker.so` fixed in #16002
- SONAME for `libdrake_ibex.so` [fixes added](https://github.com/RobotLocomotion/drake/pull/15963/files#diff-420736f108b598b2ece343dc8d4c6ff0c62aa469555d4f36263669f1883581daR357-R362) in #15963

Fresh compilation / local install confirms the `libdrake*.so` libraries all have an `SONAME` field now, closing this issue.

There are still libraries missing SONAMES, but they are **all** coming from the python bindings and probably can be safely ignored or fixed in a new issue.

```console
$ cat NOTFOUND_sonames.txt | wc -l
152
$ cat NOTFOUND_sonames.txt | grep -v site-packages | wc -l
0
$ cat NOTFOUND_sonames.txt | cut -f 1-7 -d / | uniq | sort
/tmp/drake-sonames-fixed/lib/python3.8/site-packages/director
/tmp/drake-sonames-fixed/lib/python3.8/site-packages/lcm
/tmp/drake-sonames-fixed/lib/python3.8/site-packages/pydrake
/tmp/drake-sonames-fixed/lib/python3.8/site-packages/vtkmodules
```
